src/oc_erchef/apps/oc_chef_wm/src/oc_chef_wm_policy_group_policy_rev.erl

```diff
@@ -0,0 +1,297 @@
+                    cleanup_orphaned_authz_ids(PolicyState#policy_state.created_policy_group, RequestorId, PolicyState#policy_state.policy_group_authz_id),
+                    oc_chef_wm_base:multi_auth_check_to_wm_response(MultiAuthResult)
+            end
+    end.
+
+auth_info(Req, #base_state{chef_db_context = DbContext,
+                           resource_state = #policy_state{oc_chef_policy_group_revision_association = QueryRecord} = PolicyState
+                           } = State) ->
+    %% for GET and DELETE we want to save the serialized object in here.
+    PolicyAssoc = oc_chef_policy_group_revision_association:find_policy_revision_by_orgid_name_group_name(QueryRecord, DbContext),
+    StateWithResponse = case PolicyAssoc of
+        #oc_chef_policy_group_revision_association{id = ID, serialized_object = Object} ->
+            PolicyStateWithResponseData = PolicyState#policy_state{
+                    policy_assoc_exists = ID,
+                    policy_data_for_response = Object},
+            State#base_state{resource_state = PolicyStateWithResponseData};
+        _Any -> State
+    end,
+
+    PermissionsObjects = policy_permissions_objects(wrq:method(Req), PolicyAssoc, QueryRecord, DbContext),
+    StateWithAuthzIDs = stash_permissions_objects_authz_ids(PermissionsObjects, StateWithResponse),
+    PermissionsListOrHalt = permissions_with_actions(PermissionsObjects, Req),
+
+    case PermissionsListOrHalt of
+        {halt, 404, Message} ->
+            ReqWithBody = chef_wm_util:set_json_body(Req, Message),
+            {{halt, 404}, ReqWithBody, StateWithAuthzIDs#base_state{log_msg = policy_not_found}};
+        PermissionsList ->
+            {PermissionsList, Req, StateWithAuthzIDs}
+    end.
+
+
+stash_permissions_objects_authz_ids(Halt, State) when is_tuple(Halt) ->
+    State;
+stash_permissions_objects_authz_ids([], State) ->
+    State;
+stash_permissions_objects_authz_ids([{create_in_container, _C}|Rest], State) ->
+    stash_permissions_objects_authz_ids(Rest, State);
+stash_permissions_objects_authz_ids([{policy_group,AzID}|Rest], #base_state{resource_state = PolicyState} = State) ->
+    UpdatedPolicyState = PolicyState#policy_state{policy_group_authz_id = AzID},
+    UpdatedBaseState = State#base_state{resource_state = UpdatedPolicyState},
+    stash_permissions_objects_authz_ids(Rest, UpdatedBaseState);
+stash_permissions_objects_authz_ids([{policy,AzID}|Rest], #base_state{resource_state = PolicyState} = State) ->
+    UpdatedPolicyState = PolicyState#policy_state{policy_authz_id = AzID},
+    UpdatedBaseState = State#base_state{resource_state = UpdatedPolicyState},
+    stash_permissions_objects_authz_ids(Rest, UpdatedBaseState).
+
+cleanup_orphaned_authz_ids(false, _RequestorId, _AuthzID) ->
+    ok;
+cleanup_orphaned_authz_ids(true, RequestorId, AuthzId) ->
+    oc_chef_authz:delete_resource(RequestorId, object, AuthzId),
```

Comment:
Yup, these are generated in this request. This endpoint does create or update against two container/object types, so you might have permission for one and not the other. This gets called in `forbidden` (not using the one from `wm_base`) to clean up before returning the 403.